src/Nri/Ui/RadioButton/V3.elm

```diff
@@ -0,0 +1,561 @@
+            }
+
+
+{-| Show the pennant and attach this onClick handler
+-}
+showPennant : msg -> Attribute value msg
+showPennant premiumMsg =
+    Attribute { emptyEventsAndValues | premiumMsg = Just premiumMsg } identity
+
+
+{-| Content that shows when this RadioButton is selected
+-}
+disclosure : List (Html msg) -> Attribute value msg
+disclosure childNodes =
+    Attribute { emptyEventsAndValues | disclosedContent = childNodes } identity
+
+
+{-| Customizations for the RadioButton.
+-}
+type Attribute value msg
+    = Attribute (EventsAndValues value msg) (Config -> Config)
+
+
+type alias EventsAndValues value msg =
+    { value : Maybe value
+    , selectedValue : Maybe value
+    , onSelect : Maybe (value -> msg)
+    , valueToString : Maybe (value -> String)
+    , premiumMsg : Maybe msg
+    , disclosedContent : List (Html msg)
+    }
+
+
+emptyEventsAndValues : EventsAndValues value msg
+emptyEventsAndValues =
+    { value = Nothing
+    , selectedValue = Nothing
+    , onSelect = Nothing
+    , valueToString = Nothing
+    , premiumMsg = Nothing
+    , disclosedContent = []
+    }
+
+
+{-| This is private. The public API only exposes `Attribute`.
+-}
+type alias Config =
+    { name : Maybe String
+    , teacherPremiumLevel : Maybe PremiumLevel
+    , contentPremiumLevel : Maybe PremiumLevel
+    , isDisabled : Bool
+    , showPennant : Bool
+    }
+
+
+emptyConfig : Config
+emptyConfig =
+    { name = Nothing
+    , teacherPremiumLevel = Nothing
+    , contentPremiumLevel = Nothing
+    , isDisabled = False
+    , showPennant = False
+    }
+
+
+applyConfig : List (Attribute value msg) -> Config
+applyConfig attributes =
+    List.foldl (\(Attribute _ update) config -> update config)
+        emptyConfig
+        attributes
+
+
+orExisting : (acc -> Maybe a) -> acc -> acc -> Maybe a
+orExisting f new previous =
+    case f previous of
+        Just just ->
+            Just just
+
+        Nothing ->
+            f new
+
+
+applyEvents : List (Attribute value msg) -> EventsAndValues value msg
+applyEvents =
+    List.foldl
+        (\(Attribute eventsAndValues _) existing ->
+            { value = orExisting .value eventsAndValues existing
+            , selectedValue = orExisting .selectedValue eventsAndValues existing
+            , onSelect = orExisting .onSelect eventsAndValues existing
+            , valueToString = orExisting .valueToString eventsAndValues existing
+            , premiumMsg = orExisting .premiumMsg eventsAndValues existing
+            , disclosedContent = eventsAndValues.disclosedContent ++ existing.disclosedContent
+            }
+        )
+        emptyEventsAndValues
+
+
+maybeAttr : (a -> Html.Attribute msg) -> Maybe a -> Html.Attribute msg
+maybeAttr attr maybeValue =
+    maybeValue
+        |> Maybe.map attr
+        |> Maybe.withDefault Attributes.none
+
+
+{-| View a single radio button.
+Renders nothing if the attributes list does not contain value, name, and valueToString.
```

Comment:
on the fence here where maybe instead of `String -> List (Attribute value msg) -> Html msg` we should go full circle back to `{label : String, name : String, value : value, valueToString : value -> String } -> List (Attribute value msg) -> Html msg` and only use the attributes list for non-critical fields.